### PR TITLE
[Wallet] Add dynamic notification to tray from Firebase

### LINKED ISSUE #6137: cUSD rewards consumer education
card from push notification has link to web-view with information such as: education and rewards earned so far.
Page has link to buy cUSD, but maybe if account hasn't verified, then link could be to verify instead.

### PR DESCRIPTION
**Pull request by gnardini**
### Description

Allows adding a notification to the `NotificationBox` dynamically from Firebase. The CTA opens a `WebView` with an URI received from Firebase.

Screen Shot 2021-01-06 at 02 58 33

### Other changes

Added a `darkMode` option to the notifications on the `NotificationBox`. Dynamic notifications only look good on dark mode for now since we only have one icon, but adding a different icon for normal mode could fix that in the future.

### Tested

Yes

### Related issues

- Fixes #6137

### Backwards compatibility

N/A

### CODE REVIEW

File: packages/mobile/src/home/reducers.ts [hunk 1]
```diff
@@ -1,34 +1,84 @@
...
+  cta: string
+  dismiss: string
+}
+
+export interface Notification {
+  ctaUri: string
+  darkMode: boolean
+  en: NotificationTexts
+  es: NotificationTexts
+  pt: NotificationTexts
```
Comment (jeanregisser) [inline packages/mobile/src/home/reducers.ts]: nit: grouping these with the rest makes typing slightly more difficult and error prone as we add more languages.
```suggestion
  content: { [lang: string] : NotificationTexts | undefined }
```

File: packages/mobile/src/home/NotificationBox.tsx [hunk 2]
```diff
@@ -199,6 +213,45 @@ export class NotificationBox extends React.Component<Props, State> {
       })
     }
 
+    for (const [id, notification] of Object.entries(this.props.extraNotifications)) {
+      if (!notification || notification.dismissed) {
+        continue
+      }
+      const texts = notification[
+        i18n.language.slice(0, 2) as keyof Notification
```
Comment (jeanregisser) [inline packages/mobile/src/home/NotificationBox.tsx]: I'm thinking we should use the full language tag instead so it matches what we do elsewhere for localized data.

**Review by jeanregisser — COMMENTED** [review]
Great! 👍  See some small comments below.

File: packages/mobile/src/home/reducers.ts (hunk 1 above)
Comment (gnardini) [reply packages/mobile/src/home/reducers.ts]: That sounds a lot better, I was looking for some way of doing that 😄

File: packages/mobile/src/home/NotificationBox.tsx (hunk 2 above)
Comment (gnardini) [reply packages/mobile/src/home/NotificationBox.tsx]: Hmm, on Firebase we have the keys `en`, `es` and `pt`, are you suggesting having different translations for every country? Like separating `en-US` from `en-GB`?

Comment (aaronmgdr) [reply packages/mobile/src/home/NotificationBox.tsx]: probably best to have both the more specific and generic ways work. check lang-cc first then fall back to just lang.

Comment (gnardini) [reply packages/mobile/src/home/NotificationBox.tsx]: I don't think it's likely that we'll use that, but I don't see why not have it anyways, I added the full language as default and the 2 letter code as fallback :)

**Review by jeanregisser — COMMENTED** [review]
Almost good! We also need to update the db-rules: https://github.com/celo-org/celo-monorepo/blob/23a904e9fe51b810b1fb304ca5c258c08b5071d6/packages/mobile/src/firebase/db-rules.json#L1

**Comment by gnardini** [comment]
> Almost good! We also need to update the db-rules:

I didn't know we had this! I edited them manually from the console in the past, at what point does these rules get updated to the database?

**Comment by jeanregisser** [comment]
> I didn't know we had this! I edited them manually from the console in the past, at what point does these rules get updated to the database?

Yes it's only done manually for now.

**Review by jeanregisser — APPROVED** [review]
🚀

**Comment by Lss-Ankit** [comment]
Hi @gnardini We have verified this task on latest testflight 1.8.0(40) & android internal build 1.8.0 and updated the status in sheet.

**Test plan link:** https://docs.google.com/spreadsheets/d/1phHo7MUMQNLjuJOic1TlZwoHM2T0iecT07VLoKtiqMg/edit#gid=177831960

**New Issue:**
- #6443 

**Note:** We are block to get the rewards &  not able to buy the cUSD because user should redirect to add funds page by tapping on buy cUSD button on simplex page 
However. we are using mainnet for valora here user redirecting to simplex page for buying cUSD So, we are blocked to buy the cUSD 
Can you please do the needful that how we can buy the cUSD.
Thanks!

**Pull request merged**